core/modules/installer/installer.pages.inc

```diff
@@ -650,7 +650,7 @@ function installer_browser_installation_enable_form($form, &$form_state, $projec
   foreach ($projects as $project) {
     if ($project['type'] == 'module' && isset($modules[$project['name']])) {
       foreach ($modules as $module) {
-        if (isset($module->info['project']) && isset($modules[$project['name']]->info['project']) && $modules[$project['name']]->info['project'] == $module->info['project'] && $modules[$project['name']]->name != $module->name && !$module->info['hidden']) {
+        if (isset($module->info['project']) && isset($modules[$project['name']]->info['project']) && $modules[$project['name']]->info['project'] == $module->info['project'] && $modules[$project['name']]->name != $module->name && !isset($module->info['hidden'])) {
```

Comment:
> Nice & simple fix but I'm wondering if, while we're fixing this line, we should make it easier to read/understand...? I'm not sure what the coding standards say about long `if` conditions with multiple `&&`'s, but any ideas how to make this more readable? Spreading over more lines, using multiple `if`'s...?

Both d.org and our coding standards have this covered: https://api.backdropcms.org/php-standards#line-length

> Control structure conditions should also NOT attempt to win the _Most Compact Condition In Least Lines Of Code Award™_: ... Instead, it is recommended practice to split out and prepare the conditions separately, which also permits documenting the underlying reasons for the conditions:

@BWPanda something like this then?:

```suggestion
        $module_has_project = isset($module->info['project']) && isset($modules[$project['name']]->info['project']);
      $module_is_submodule = $modules[$project['name']]->info['project'] == $module->info['project'] && $modules[$project['name']]->name != $module->name;
      $module_not_hidden = !isset($module->info['hidden']);
      foreach ($modules as $module) {
        if ($module_has_project && $module_is_submodule && $module_not_hidden) {
```

Reply:
Something like that yes, but not exactly that...

Here's the original conditional expanded out for easier reading and numbered for easier reference:

```php
if (
  isset($module->info['project']) // #1
   &&
  isset($modules[$project['name']]->info['project']) // #2
   &&
  $modules[$project['name']]->info['project'] == $module->info['project'] // #3
   &&
  $modules[$project['name']]->name != $module->name // #4
   &&
  !$module->info['hidden'] // #5
) {
```

It seems to me that conditions #1-3 are related, and conditions #4 & #5 are seperate (i.e. 3 'groups' of conditions).

Some thoughts:

- It seems as if in order to check #3, you first have to make sure the individual components exist (hence #1 & #2). But is this necessary?
- #4 doesn't bother checking if its components exist first (but should it?)
- If `$module->info['hidden']` is either `TRUE`, `FALSE`, or just not set, then I think the easiest check here would be: `empty($module->info['hidden'])` (that'll make sure it's either `FALSE` or not set at all)

Reply:
Ad 1 and 2: Regarding the two isset() checks: yes, is necessary, because "project" may be set or not. - Or am I missing something?

Ad 4: "name" must be set, no matter what.
 
Ad 5 and empty() vs isset() to catch a possible (though unlikely) FALSE - good point.